app/src/main/java/framgia/com/ichat/screen/chat/ChatAdapter.java

```diff
@@ -0,0 +1,149 @@
+package framgia.com.ichat.screen.chat;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import framgia.com.ichat.ApplicationGlideModule;
+import framgia.com.ichat.GlideApp;
+import framgia.com.ichat.R;
+import framgia.com.ichat.data.model.Message;
+import framgia.com.ichat.utils.DateTime;
+
+public class ChatAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
+    private static final int SENT = 0;
+    private static final int RECEIVE = 1;
+    private Context mContext;
+    private List<Message> mMessages;
+    private LayoutInflater mInflater;
+    private String mId;
+    private OnMessageItemClickListener mOnClickListener;
+
+    public ChatAdapter(Context context, String id,
+                       OnMessageItemClickListener onClickListener) {
+        mContext = context;
+        mMessages = new ArrayList<>();
+        mInflater = LayoutInflater.from(mContext);
+        mId = id;
+        mOnClickListener = onClickListener;
+    }
+
+    public void addData(Message message) {
+        mMessages.add(message);
+        notifyItemInserted(mMessages.size() - 1);
+    }
+
+    @Override
+    public int getItemViewType(int position) {
+        if (mMessages.get(position).getSenderId().equalsIgnoreCase(mId)) {
+            return SENT;
+        }
+        return RECEIVE;
+    }
+
+    @NonNull
+    @Override
+    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
+        switch (i) {
+            case SENT:
+                return new ViewHolderMessageSent(mInflater.inflate(R.layout.item_message_sent,
+                        viewGroup, false));
+            case RECEIVE:

```

Comment:
```suggestion
            case RECEIVE:
            default:
```
viết ntn tránh duplicate dòng 60 và 63